cstypo/parser.py

```diff
@@ -185,10 +185,10 @@ def parse_dates(self, text):
         """
 
         with_year = re.compile(r'(?<!\d)(\d{1,2}\.) (\d{1,2}\.) (\d\d)')
-        substituted = with_year.sub(r'\1\u00a0\2\u00a0\3', text)
+        substituted = self.sub(with_year, r'\1\u00a0\2\u00a0\3', text)
```

Comment:
```suggestion
        substituted = with_year.sub('\\1\u00a0\\2\u00a0\\3', text)
```
Your solution will not work since the character escape sequence in raw string will not be interpreted.